client/client.go

```diff
@@ -303,8 +303,9 @@ func (r *NotaryRepository) RemoveTarget(targetName string) error {
 	return nil
 }
 
-// ListTargets lists all targets for the current repository
-func (r *NotaryRepository) ListTargets() ([]*Target, error) {
+// ListTargets lists all targets for the current repository. The list of
+// roles should be passed in order from lowest to highest priority
+func (r *NotaryRepository) ListTargets(roles ...string) ([]*Target, error) {
```

Comment:
Order dependent on arguments is weird, but plenty of things are order dependent on values. I could have written this as `ListTargets(roles []string)` but if we want to support not having to pass roles, which we well might, this format is easier for passing no roles (i.e. `ListTargets()` vs `ListTargets(nil)`).

TUF's spec talks about using the ordering of delegations to determine priority. We probably don't want to do exactly that as we want to manipulate priority across multiple levels (like putting "targets/releases" as a high priority than just "targets"). I've just realized I didn't update the function quite right, but the intent is to load the highest priority metadata last, having it overwrite the lower priority data.

Updating now